### PR TITLE
Provide a context that manages access to a Dropwizard app.

### PR DESCRIPTION
**Pull request by ojacobson**
If `app.url` is set, tests using this context will talk to the app at that
URL.

Otherwise, tests using this context will spin up a temporary app for each
test, and shut it down at the end of each test.

### CODE REVIEW

**Comment by ojacobson** [comment]
Needs tests, but I'm _way_ too fried to write them tonight.

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        assertThat(
+                applicationContext.rules(),
+                instanceOf(DropwizardAppRule.class));
+        applicationContext
+                .rules()
+                .apply(
+                        new Statement() {
+                            @Override
+                            public void evaluate() throws Throwable {
+                                assertThat(
```
Comment (ojacobson) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java]: @lewisd32 Curious what you think of this technique.

Comment (lewisd32) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java]: I like the overall approach, but I feel a little funny about the `startswith`.

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/StringsTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Test
+    public void findLastNonSlash() {
+        assertThat(Strings.findLastNonSlash(""), equalTo(0));
+        assertThat(Strings.findLastNonSlash("/"), equalTo(0));
+        assertThat(Strings.findLastNonSlash("////"), equalTo(0));
+        assertThat(Strings.findLastNonSlash("foo"), equalTo(3));
+        assertThat(Strings.findLastNonSlash("/foo/"), equalTo(4));
+        assertThat(Strings.findLastNonSlash("foo/bar/////"), equalTo(7));
+
```
Comment (lewisd32) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/StringsTest.java]: One extra blank line, if I'm being fussy

File: unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/Strings.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                return i;
+        return 0;
+    }
+
+    public static int findFirstNonSlash(String s) {
+        for (int i = 0; i < s.length(); ++i)
+            if (s.charAt(i) != '/')
+                return i;
+        return s.length();
+    }
```
Comment (lewisd32) [inline unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/Strings.java]: Do the above two methods need to be public? I can only find them being called internally, and by tests.  Do they even need to be tested separately from the `strip*` methods?

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java (hunk 1 above)
Comment (ojacobson) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/ApplicationContextTest.java]: Sold.

File: unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/Strings.java (hunk 3 above)
Comment (ojacobson) [inline unacceptable-dropwizard/src/main/java/io/github/unacceptable/dropwizard/context/Strings.java]: They do not.

File: unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/StringsTest.java (hunk 2 above)
Comment (ojacobson) [inline unacceptable-dropwizard/src/test/java/io/github/unacceptable/dropwizard/context/StringsTest.java]: Please, be fussy!

**Comment by ojacobson** [comment]
I've re-pushed this with commentary.

**Pull request merged**